Mount Updates

rosenhauer opened the pull request:
baseMounts->boxMounts
Cleaned up some of the echo statements - removed some pure debugging ones.

Updated to show the predefined shapes and masks in the console with the other global variables.

![image](https://github.com/mrWheel/YAPP_Box/assets/33765928/e1c73a23-8d87-4ed8-a11c-efb27edecd4f)

Fixed minor fillet issues with buttons (wasn't suppressing them if told to)

Mounts:
Renamed vector name.
Noticed mount positioning wasn't consistent - fixed.
Saw that with roundRadius > mount thickness fillet didn't touch case properly - fixed.
Fillet could extent into screw opening - fixed.

Added option for negative mount width.  This rotates the slot 90 degrees so it is perpendicular to the case instead of parallel.
![image](https://github.com/mrWheel/YAPP_Box/assets/33765928/c969b12d-ec54-4c64-a787-eb47fca24cb2)

See new boxMounts demo for examples.

mrWheel commented:
@rosenhauer,

## I have found something in `RidgeExtDemo_v30`:
```
ECHO: "YAPP=========================================="
[WARNING: Unable to convert translate([undef, 75, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 85, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 98, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 25, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 45, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 38, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 75, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 85, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 98, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 25, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 45, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
[WARNING: Unable to convert translate([undef, 38, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad) 
Compiling design (CSG Products generation)...
```
## I removed some old files and added some new files to the repo

## Also:
I sometimes want to add pcbStands and/or connectors from an image like this:

![image](https://github.com/mrWheel/YAPP_Box/assets/5585427/6c771f30-85a4-41f2-b518-95f4f84ab103)

As you can see all measurements are done from pcb[0,0,0] (as I have learned at school), but it is no longer possible to enter the coordinates for the stands with these values ... (or am I doing something wrong?)
In pré v3.0 I could enter this:
```
pcbStands =     
[
     [3.6,  3, 2, yappBoth, yappPin]                     // back-left
   , [3.6,  pcbWidth-3, 2, yappBoth, yappHole]           // back-right
   , [pcbLength-3,  7.5, 2, yappBoth, yappHole]          // front-left
   , [pcbLength-3, pcbWidth-3, 2, yappBoth, yappPin]     // front-right
 ];
```
but with v3.0 I have to enter this:
```
pcbStands =     
[
    [3.6, 3, 2, yappBoth, yappPin, yappBackLeft]      // back-left
   ,[3.6, 3.4, 2, yappBoth, yappPin, yappFrontRight]  // back-right
   ,[3.6, 7, 2, yappBoth, yappHole, yappFrontLeft]    // front-left
   ,[3.6, 3.4, 2, yappBoth, yappHole, yappBackRight]  // front-right

];     
```
 You have to calculate all stands as if the origen is pcb[0,0,0].
In this example it is not that difficult to translate, but there is no more relation with the original drawing.

rosenhauer commented:
On the RidgeExtDemo_v30 that is the old version.  There is a new
YAPP_RidgeExtDemo_v30.scad that should be used.

on the pcbStands.  The 6th param (yappBackLeft,yappBackRight...) tells it
what to use as the origin or origins. So you can do it as before if you
specify the prime origin (yappBackLeft) as the origin for each stand:

pcbStands =
[
     [3.6,  3, 2, yappBoth, yappPin, yappBackLeft]
// back-left

   , [3.6,  pcbWidth-3, 2, yappBoth, yappHole, yappBackLeft]
// back-right

   , [pcbLength-3,  7.5, 2, yappBoth, yappHole, yappBackLeft]
// front-left

   , [pcbLength-3, pcbWidth-3, 2, yappBoth, yappPin, yappBackLeft]
// front-right

 ];



I changed it to have the default the prime origin instead of
yappAllCorners.  Now it will work like before.  Will push update in a few
min after I test it.



On Tue, Dec 12, 2023 at 8:06 AM mrWheel ***@***.***> wrote:

> @rosenhauer <https://github.com/rosenhauer>,
> I have found something in RidgeExtDemo_v30:
>
> ECHO: "YAPP=========================================="
> [WARNING: Unable to convert translate([undef, 75, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 85, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 98, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 25, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 45, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 38, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 75, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 85, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 98, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 25, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 45, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> [WARNING: Unable to convert translate([undef, 38, 0]) parameter to a vec3 or vec2 of numbers in file library/YAPPgenerator_v30.scad, line 1949](https://github.com/mrWheel/YAPP_Box/pull/1949,/Users/WillemA/Documents/openSCAD/YAPP_Box/library/YAPPgenerator_v30.scad)
> Compiling design (CSG Products generation)...
>
> I removed some old files and added some new files to the repo Also:
>
> I sometimes want to add pcbStands and/or connectors from an image like
> this:
>
> image.png (view on web)
> <https://github.com/mrWheel/YAPP_Box/assets/5585427/6c771f30-85a4-41f2-b518-95f4f84ab103>
>
> As you can see all measurements are done from pcb[0,0,0] (as I have
> learned at school), but it is no longer possible to enter the coordinates
> for the stands with these values ... (or am I doing something wrong?)
> In pré v3.0 I could enter this:
>
> pcbStands =
> [
>      [3.6,  3, 2, yappBoth, yappPin]                     // back-left
>    , [3.6,  pcbWidth-3, 2, yappBoth, yappHole]           // back-right
>    , [pcbLength-3,  7.5, 2, yappBoth, yappHole]          // front-left
>    , [pcbLength-3, pcbWidth-3, 2, yappBoth, yappPin]     // front-right
>  ];
>
> but with v3.0 I have to enter this:
>
> pcbStands =
> [
>     [3.6, 3, 2, yappBoth, yappPin, yappBackLeft]      // back-left
>    ,[3.6, 3.4, 2, yappBoth, yappPin, yappFrontRight]  // back-right
>    ,[3.6, 7, 2, yappBoth, yappHole, yappFrontLeft]    // front-left
>    ,[3.6, 3.4, 2, yappBoth, yappHole, yappBackRight]  // front-right
>
> ];
>
> You have to calculate all stands as if the origen is pcb[0,0,0].
> In this example it is not that difficult to translate, but there is no
> more relation with the original drawing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrWheel/YAPP_Box/pull/45#issuecomment-1852104116>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBTUKBBMRCGUMQW3IDXX6TYJBQFLAVCNFSM6AAAAABAQLR2NOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJSGEYDIMJRGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Dave

rosenhauer commented:
>
> I found this strange artifact that I can't get rid of...


I've not seen that before.  I messed around with different wallThickness
and RoundRadius and can't reproduce it.   Can you send the full script that
generated that?

Could you add a value
> default = undef;
> deflt = undef;
> So, instead of using undef we can write default in parameter list.


To be consistent would you want to use yappDefault?

On Wed, Dec 13, 2023 at 8:42 AM mrWheel ***@***.***> wrote:

> @rosenhauer <https://github.com/rosenhauer>,
>
> I found this strange artifact that I can't get rid of...
>
> boxMounts   =
> [
>    [0,           3.5, 15, 3, yappLeft, yappCenter]
>   ,[shellLength, 3.5, 15, 3, yappLeft, yappCenter]
>   ,[0,           3.5, 15, 3, yappLeft, yappLid, yappCenter]
>   ,[shellLength, 3.5, 15, 3, yappLeft, yappLid, yappCenter]
> ];
>
> Screenshot.2023-12-13.at.11.21.04.png (view on web)
> <https://github.com/mrWheel/YAPP_Box/assets/5585427/c87d7f10-883d-4f6a-b305-4267798eeb24>
> default = undef;
>
> Could you add a value
>
> default = undef;
> deflt   = undef;
>
> So, instead of using undef we can write default in parameter list.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mrWheel/YAPP_Box/pull/45#issuecomment-1854044819>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIBTUKF26TG5VXNVCE5EOULYJG5F3AVCNFSM6AAAAABAQLR2NOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJUGA2DIOBRHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Dave

mrWheel commented:
@rosenhauer,
You are amazing!
Problem is gone!
